dataplane/snappi/sai_dataplane.py

```diff
@@ -0,0 +1,334 @@
+            request.port_name = name
+            pcap_bytes = self.api.get_capture(request)
+
+            cap_dict[name] = []
+            for ts, pkt in dpkt.pcap.Reader(pcap_bytes):
+                cap_dict[name].append(list(pkt))
+
+        return cap_dict
+
+    def start_traffic(self):
+        """ Start traffic flow(s) which are already configured.
+        """
+        ts = self.api.transmit_state()
+        ts.state = ts.START
+        logging.info('Start traffic')
+        res = self.api.set_transmit_state(ts)
+        assert self.api_results_ok(res), res
+
+    def stop_traffic(self):
+        """ Stop traffic flow(s) which are already configured.
+        """
+        ts = self.api.transmit_state()
+        ts.state = ts.STOP
+        logging.info('Stop traffic')
+        res = self.api.set_transmit_state(ts)
+        assert self.api_results_ok(res), res
+
+    # PAUSE api not supported by Ixia-C
+    def pause_traffic(self):
+        """ Pause traffic flow(s) which are already configured.
+        """
+        ts = self.api.transmit_state()
+        ts.state = ts.PAUSE
+        logging.info('Pause traffic')
+        res = self.api.set_transmit_state(ts)
+        assert self.api_results_ok(res), res
+
+    def is_traffic_stopped(self, flow_names=[]):
+        """
+        Returns true if traffic in stop state
+        """
+        fq = self.api.metrics_request()
+        fq.flow.flow_names = flow_names
+        metrics = self.api.get_metrics(fq).flow_metrics
+        return all([m.transmit == "stopped" for m in metrics])
+
+    def get_all_stats(self, print_output=True):
+        """
+        Returns all port and flow stats
+        """
+        print("Fetching all port stats ...")
+        request = self.api.metrics_request()
+        request.choice = request.PORT
+        request.port
+        port_results = self.api.get_metrics(request).port_metrics
+        if port_results is None:
+            port_results = []
+
+        # print("Fetching all flow stats ...")
+        # request = self.api.metrics_request()
+        # request.choice = request.FLOW
+        # request.flow
+        # flow_results = self.api.get_metrics(request).flow_metrics
+        # if flow_results is None:
+        #     flow_results = []
+
+        # if print_output:
+        #     print_stats(port_stats=port_results, flow_stats=flow_results)
+
+        return port_results
+
+    def stats_ok(self, packets):
+        """
+        Returns true if stats are as expected, false otherwise.
+        """
+        _, flow_stats = self.get_all_stats()
+
+        flow_rx = sum([f.frames_rx for f in flow_stats])
+        return flow_rx == packets
+
+    def add_flow(self,
+                 name,
+                 packet_count=1,
+                 seconds_count=0,
+                 pps=10,
+                 force_pps=False
+                 ):
+
+        name = name or f"flow_{time.time()}"
+        flow = self.configuration.flows.flow(name=name)[-1]
```

Comment:
would suggest adding new flow using new `add` construct.

https://github.com/open-traffic-generator/conformance/blob/main/features/flows/headers/udp/test_udp_header.py#L52

Reply:
Added to backlog.